busybox: fix compatibility with BUSYBOX_CONFIG_INSTALL_NO_USR

M95D opened the pull request:
Fixes two bugs that manifest when  BUSYBOX_CONFIG_INSTALL_NO_USR is selected.
See each commit for details.

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

Commit:
busybox: fix compatibility with BUSYBOX_CONFIG_INSTALL_NO_USR

If busybox is built with BUSYBOX_CONFIG_INSTALL_NO_USR, crond and ntpd will be installed in /sbin, not in /usr/sbin. This breaks cron and ntpd startup scripts.

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

M95D commented:
Why is busybox installed in /usr? It creates problems like [overwriting full programs with bb applets when updating busybox](https://forum.openwrt.org/t/limit-busybox-applets-to-bin-and-sbin/71915). Isn't /usr for user programs on all unix systems? Busybox is a core package. It dosn't belong there.
New option:
* Move busybox by default to /bin and /sbin and make BUSYBOX_CONFIG_INSTALL_NO_USR permanently enabled.

M95D commented:
When busybox is built with BUSYBOX_CONFIG_INSTALL_NO_USR, /usr/bin/env applet is created in /bin/env instead. This breaks Luci.
Added a separate fix that creates the symlink as /usr/bin/env.

M95D commented:
I think this is ready for commit. I found no other incompatibilities.

M95D commented:
@jow-  @hauke 
Please review.

hauke commented on package/utils/busybox/Makefile:
```diff
@@ -155,7 +163,7 @@ ifneq ($(CONFIG_BUSYBOX_$(BUSYBOX_SYM)_NTPD),)
 	$(INSTALL_DATA) ./files/ntpd.capabilities $(1)/etc/capabilities/ntpd.json
 	$(INSTALL_DATA) ./files/ntpd_acl.json $(1)/usr/share/acl.d/ntpd.json
 endif
-	-rm -rf $(1)/lib64
```
Why is this remove not needed any more?

hauke commented on package/utils/busybox/Makefile:
```diff
@@ -138,8 +138,16 @@ define Build/Compile
-	$(INSTALL_DIR) $(1)/etc/init.d
+
+ifneq ($(CONFIG_BUSYBOX_$(BUSYBOX_SYM)_INSTALL_NO_USR),)
+	mkdir -p $(PKG_INSTALL_DIR)/usr/bin
```
Why don't you use `$(INSTALL_DIR) $(1)/usr/bin` ?

hauke commented on package/utils/busybox/Makefile:
```diff
@@ -138,8 +138,16 @@ define Build/Compile
+	$(INSTALL_DIR) $(1)/bin
+	$(INSTALL_DIR) $(1)/sbin
+endif
+	$(INSTALL_DIR) $(1)/usr/bin
```
Why is `$(INSTALL_DIR) $(1)/usr/bin` added here?

Commit:
busybox: fix missing /usr/bin/env

When busybox is built with BUSYBOX_CONFIG_INSTALL_NO_USR,
it will install "env" applet in /bin, not /usr/bin,
and Luci scripts will fail. This fix creates the missing symlink.

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

M95D commented:
@hauke Polite ping.

Commit:
base-files: ipcalc.sh: fix path to awk in shebang

If firmware is built with BUSYBOX_CONFIG_INSTALL_NO_USR,
busybox awk will be installed in /bin, not /usr/sbin.
Fix this by using /usr/bin/env (already used by Luci) and
awk without path.

Fixes bug #12089

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

hauke commented on package/utils/busybox/files/sysntpd:
```diff
@@ -4,7 +4,7 @@
 
 USE_PROCD=1
-PROG=/usr/sbin/ntpd
+PROG=ntpd
```
jail does not allow this.
procd fails like this:
````
Sun Mar  5 23:08:39 2023 user.err : jail: ntpd is not an absolute path
````

M95D replied:
How about this?
PROG=$(which ntpd)

If not, BUSYBOX_CONFIG_INSTALL_NO_USR should be permanently removed.
